### PR TITLE
Add tool call info in API responses if requested

### PR DESCRIPTION
**Pull request by RobertSzefler**

### CODE REVIEW

File: server.py [hunk 1]
```diff
@@ -55,14 +57,26 @@ def investigate_issue(request: InvestigateRequest):
...
-    return {
-        "analysis": ai.investigate(
-            issue,
-            # TODO prompt should probably be configurable?
-            prompt=load_prompt("builtin://generic_investigation.jinja2"),
-            console=console,
-        ).result
+    investigation = ai.investigate(
+        issue,
+        # TODO prompt should probably be configurable?
```
Comment (aantn) [inline server.py:88]: Yes, I think it is worth supporting an override (optional though)
@arikalon1 do you agree?

**Review by aantn — APPROVED** [review]
LGTM. Left one comment not directly related to this PR. You can merge regardless of that comment.

File: server.py (hunk 1 above)
Comment (arikalon1) [reply server.py:88]: yeah, I agree
We should allow an optional override
@RobertSzefler can you please add it?

Comment (RobertSzefler) [reply server.py:88]: added

**Pull request merged**